extension/observer/ecstaskobserver/factory.go

```diff
@@ -0,0 +1,54 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ecstaskobserver
+
+import (
+	"context"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config"
+	"go.opentelemetry.io/collector/extension/extensionhelper"
+)
+
+const (
+	typeStr config.Type = "ecs_task_observer"
+)
+
+// NewFactory creates a factory for ECSTaskObserver extension.
+func NewFactory() component.ExtensionFactory {
+	return extensionhelper.NewFactory(
+		typeStr,
+		createDefaultConfig,
+		createExtension)
+}
+
+func createDefaultConfig() config.Extension {
+	cfg := defaultConfig()
+	return &cfg
+}
+
+func createExtension(
+	_ context.Context,
+	params component.ExtensionCreateSettings,
+	cfg config.Extension,
+) (component.Extension, error) {
+	obsCfg := cfg.(*Config)
+	e := &ecsTaskObserver{
+		config: obsCfg,
+		logger: params.Logger,
```

Comment:
We recommend to pass the entire "Telemetry".

Reply:
updated